connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSinkTaskTest.java

```diff
@@ -636,6 +642,180 @@ public void run() {
         PowerMock.verifyAll();
     }
 
+    // Verify that when commitAsync is called but the supplied callback is not called by the consumer before a
+    // rebalance occurs, the async callback does not reset the last committed offset from the rebalance.
+    // See KAFKA-5731 for more information.
+    @Test
+    public void testCommitWithOutOfOrderCallback() throws Exception {
```

Comment:
Yeah, maybe the test is doing a few things that strictly speaking aren't necessary. For example, the last iteration may not be necessary, but it was only through that following iteration would [this logic](https://github.com/apache/kafka/pull/3662/files/be34cf05fe6956a24b77b35a689bda7b55e7de12#diff-9d27e74bcdc892150367aed9a4cf499eR329) be used after the rebalance. The test is, after all, verifying that a commit, a rebalance, and another commit all work correctly.